datafusion/core/src/physical_plan/file_format/mod.rs

```diff
@@ -393,6 +393,7 @@ impl SchemaAdapter {
 
     /// Map projected column indexes to the file schema. This will fail if the table schema
     /// and the file schema contain a field with the same name and different types.
+    #[allow(dead_code)]
```

Comment:
The fact this is marked as dead code seems not quite right to me

Reply:
I think this unused methods can be removed now

Reply:
fixed